File: C _ C++/C++/CP/SingleOperationPart2.c
#include <stdio.h>
#include <math.h>
int main()
{
    int t;
    scanf("%d", &t);
    while(t--)
    {
        int n, count=0, value = 0;
        scanf("%d", &n);
        char str[n];
        scanf("%s", str);
        for (int i = n-1; i >= 0; i--)
        {
            if (str[i] == '1')
            {
                value += pow(2, count);
            }
            count += 1;
        }
        int y, min = value;
        for (int i = 1; i <= n ; i++)
        {
            int temp = value/pow(2, i);
            int tem = value^temp;
            if (min >= tem)
            {
                min = tem;
                y = i;
            }
        }
        printf("%d\n", y);
    }
    return 0;
}

File: Python/LeetCode/Challenges/Wrong/SingleOperationPart2.py
def dostuff():
    n = int(input())
    s = input()
    value = 0
    for i in range(n):
        if s[len(s)-i-1] == '1':
            value += 2**i
    min = value
    for i in range(1, n+1):
        temp = int(value/(2**i))
        tem = value^temp
        if min >= tem:
            min = tem
            y = i
    print(y)
    return 0

def main():
    for _ in range(int(input())):
        dostuff()
    return 0

if __name__ == "__main__":
    main()